cdc/owner/changefeed.go

```diff
@@ -776,20 +791,38 @@ func (c *changefeed) cleanupChangefeedServiceGCSafePoints(ctx cdcContext.Context
 func (c *changefeed) preflightCheck(captures map[model.CaptureID]*model.CaptureInfo) (ok bool) {
 	ok = true
 	if c.state.Status == nil {
-		c.state.PatchStatus(func(status *model.ChangeFeedStatus) (*model.ChangeFeedStatus, bool, error) {
-			if status == nil {
-				status = &model.ChangeFeedStatus{
-					// changefeed status is nil when the changefeed has just created.
-					ResolvedTs:   c.state.Info.StartTs,
-					CheckpointTs: c.state.Info.StartTs,
-					AdminJobType: model.AdminNone,
+		// complete the changefeed status when it is just created.
+		c.state.PatchStatus(
+			func(status *model.ChangeFeedStatus) (*model.ChangeFeedStatus, bool, error) {
+				if status == nil {
+					status = &model.ChangeFeedStatus{
+						// changefeed status is nil when the changefeed has just created.
+						ResolvedTs:        c.state.Info.StartTs,
+						CheckpointTs:      c.state.Info.StartTs,
+						MinTableBarrierTs: c.state.Info.StartTs,
+						AdminJobType:      model.AdminNone,
+					}
+					return status, true, nil
 				}
-				return status, true, nil
-			}
-			return status, false, nil
-		})
+				return status, false, nil
+			})
+		ok = false
+	} else if c.state.Status.MinTableBarrierTs == 0 {
+		// complete the changefeed status when the TiCDC cluster is
+		// upgraded from an old version(less than v6.7.0).
+		c.state.PatchStatus(
+			func(status *model.ChangeFeedStatus) (*model.ChangeFeedStatus, bool, error) {
+				if status != nil {
+					if status.MinTableBarrierTs == 0 {
```

Comment:
```suggestion
```

Reply:
It should not be changed if it is not 0.